frontend/class-frontend.php

```diff
@@ -907,7 +907,7 @@ private function generate_canonical() {
 			if ( is_search() ) {
 				$search_query = get_search_query();
 
-				if ( ! empty( $search_query ) ) {
+				if ( ! empty( $search_query ) && ! preg_match( '|^page/\d+$|', $search_query ) ) {
```

Comment:
Could you please add a comment to give some context why this test is added and what cases it is intended for?